From 43b0e88947b170d335d1a15e98f63f7aff47959a Mon Sep 17 00:00:00 2001
From: Greg Sjaardema <gsjaardema@gmail.com>
Date: Wed, 19 Mar 2025 09:57:50 -0600
Subject: [PATCH] CGNS: Allow more liberal version matching

The main difference between CGNS-3.X and CGNS-4.X is the handling
of polyhedral elements.  If the cgns file does not contain those
elements, then the library should be able to read the file correctly.

We have modified the library version from the original 4.X.Y to
be 3.99 so that files written by this version of the CGNS library
(assuming they don't use polyhedral elements) will be readable by
applications that are still using CGNS-3.X
---
 src/cgnslib.c | 9 +++++----
 src/cgnslib.h | 7 +++++++
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/src/cgnslib.c b/src/cgnslib.c
index d81bbee..d890e70 100644
--- a/src/cgnslib.c
+++ b/src/cgnslib.c
@@ -587,12 +587,13 @@ int cg_open(const char *filename, int mode, int *fn)
         /* This code allows reading versions newer than the lib,
                as long as the 1st digit of the versions are equal */
             if ((cg->version / 1000) > (CGNSLibVersion / 1000)) {
-                cgi_error("A more recent version of the CGNS library created the file. Therefore, the CGNS library needs updating before reading the file '%s'.",filename);
-                return CG_ERROR;
+	      cgi_warning("A more recent version of the CGNS library created the file. Assuming it is OK to read, but may have problems  (%d vs %d).",
+			  cg->version, CGNSLibVersion);
             }
             /* warn only if different in second digit */
-            if ((cg->version / 100) > (CGNSLibVersion / 100)) {
-                cgi_warning("The file being read is more recent that the CGNS library used");
+	    else if ((cg->version / 100) > (CGNSLibVersion / 100)) {
+	      cgi_warning("The file being read is more recent than the CGNS library used (%d vs %d).",
+			  cg->version, CGNSLibVersion);
             }
         }
 #if CG_SIZEOF_SIZE == 32
diff --git a/src/cgnslib.h b/src/cgnslib.h
index 018797b..722fef7 100644
--- a/src/cgnslib.h
+++ b/src/cgnslib.h
@@ -51,6 +51,13 @@
 #define CGNS_COMPATVERSION 2540
 #define CGNS_COMPATDOTVERS 2.54
 
+/* KLUGE to allow older library to read newer files... */
+/* See also the change in cgnslib.c */
+#undef CGNS_VERSION
+#undef CGNS_DOTVERS
+#define CGNS_VERSION 3990
+#define CGNS_DOTVERS 3.99
+
 #include "cgnstypes.h"
 
 #if CG_BUILD_SCOPE
-- 
2.39.5 (Apple Git-154)

